src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -807,5 +808,196 @@ void VerifyValidEntry(ZipArchiveEntry entry, string expectedName, int expectedMi
+            // compressed size
+            0xff, 0xff, 0xff, 0xff,
+            // UNcompressed size
+            0xff, 0xff, 0xff, 0xff,
+            // file name length
+            0x08, 0x00,
+            // extra field length
+            0x38, 0x00,
+            // filename
+            0x66, 0x69, 0x6c, 0x65, 0x2e, 0x74, 0x78, 0x74,
+            // -----Zip64 extra field tag
+            0x01, 0x00,
+            // size of extra field block
+            0x10, 0x00,
+                    // 8 byte Zip64 UNcompressed size, index 42
+                    0x04, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                    // 8 byte Zip64 compressed size, index 50
+                    0x06, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            // ----- NTFS extra field tag
+            0x0a, 0x00,
+            // size of extra field block
+            0x20, 0x00,
+                    // reserved
+                    0x00, 0x00, 0x00, 0x00,
+                    // tag #1
+                    0x01, 0x00,
+                    // size of tag #1
+                    0x18, 0x00,
+                            // Mtime, CTime, Atime
+                            0xa8, 0xb1, 0xf6, 0x61, 0x25, 0x18, 0xd7, 0x01,
+                            0xa8, 0xb1, 0xf6, 0x61, 0x25, 0x18, 0xd7, 0x01,
+                            0xa8, 0xb1, 0xf6, 0x61, 0x25, 0x18, 0xd7, 0x01,
+            // -------------
+            // Data!
+            0x2b, 0x49, 0x2d, 0x2e, 0x01, 0x00,
+            // -------- Central directory signature 0x02014b50
+            0x50, 0x4b, 0x01, 0x02,
+            // version made by 4.5
+            0x2d, 0x00,
+            // version to extract 4.5
+            0x2d, 0x00,
+            // general purpose flags
+            0x00, 0x08,
+            // Deflate
+            0x08, 0x00,
+            // Last mod file time
+            0x17, 0x9b,
+            // Last mod date
+            0x6d, 0x52,
+            // CRC32
+            0x0c, 0x7e, 0x7f, 0xd8,
+            // 4 byte compressed size, index 120 (-1 indicates refer to Zip64 extra field)
+            0xff, 0xff, 0xff, 0xff,
+            // 4 byte UNcompressed size, index 124 (-1 indicates refer to Zip64 extra field)
+            0xff, 0xff, 0xff, 0xff,
+            // file name length
+            0x08, 0x00,
+            // extra field length
+            0x44, 0x00,
+            // file comment length
+            0x00, 0x00,
+            // disk number start (-1 indicates refer to Zip64 extra field)
+            0x00, 0x00,
+            // internal file attributes
+            0x00, 0x00,
+            // external file attributes
+            0x00, 0x00, 0x00, 0x00,
+            // relative offset of local header (-1 indicates refer to Zip64 extra field)
+            0x00, 0x00, 0x00, 0x00,
+            // file name
+            0x66, 0x69, 0x6c, 0x65, 0x2e, 0x74, 0x78, 0x74,
+            // extra field, content similar to before
+            0x01, 0x00,
+            0x1c, 0x00,
```

Comment:
This 'zip64 extra field' in the in the CD is the one triggering the assert -- it expected only the file sizes, but the two other fields are here also. The 'zip64 extra field' in the local file header is the one that the code ends up using the values from. In this example, that second extra field doesn't have the other two, uh, fields.